### PR TITLE
Move call of migration routine from event `OnAddOnLoad` to `OnPlayerActivated`

### PR DESCRIPTION
**Pull request by Zeratoxx**
call of migration routine at event `OnAddOnLoad` is not correct.  
it should be at event `OnPlayerActivated`.

### CODE REVIEW

**Comment by Rhyono** [comment]
Is the double ...Done intended? If so, it didn't look consistent.

**Comment by Zeratoxx** [comment]
Yes, it is intended. Here's the issue: with the last update, `jewelryIdSwapMigrationAlreadyDone` got created and set to true.
With the second "Done", I kind of reset the var. I chose "DoneDone" because it's a somewhat common term in IT 🤷‍♂️

File: CraftStore_Events.lua [hunk 1]
```diff
@@ -175,6 +175,20 @@ function CS.OnReticleHiddenUpdate(eventCode,hidden)
...
 
 function CS.OnPlayerActivated(eventCode,initial)
+  if CS.Debug then zo_callLater(function() CHAT_ROUTER:AddSystemMessage("CS.Account.crafting.jewelryIdSwapMigrationAlreadyDone: "..tostring(CS.Account.crafting.jewelryIdSwapMigrationAlreadyDone)) end, 50) end
+  if CS.Account.crafting.jewelryIdSwapMigrationAlreadyDoneDone == nil or CS.Account.crafting.jewelryIdSwapMigrationAlreadyDoneDone == false then
+    zo_callLater(
+            function()
+                CHAT_ROUTER:AddSystemMessage("[CraftStore] The internal indices of the game for jewelry (rings and necklaces) got swapped by ZOS. CraftStore need to migrate its SavedVariables once. Starting...")
+            end, 50)
+    CS.MigrateJewelryIdSwap()
+    CS.Account.crafting.jewelryIdSwapMigrationAlreadyDone = true
```
Comment (Rhyono) [inline CraftStore_Events.lua:185]: Shouldn't this be DoneDone as well?

Comment (Zeratoxx) [reply CraftStore_Events.lua:185]: Oops. You're totally right.. My head was in the sky it seems.

Comment (Zeratoxx) [reply CraftStore_Events.lua]: Changed

**Pull request merged**